### ISSUE TITLE
Wheels can not build project on mac m2 with cohere version `4.18.0`

### ISSUE THREAD

**Issue by Alavi1412** [cohere-ai/cohere-python#270]
The new release `4.18.0` has some issue with mac m2 on installation. I am receiving this error while installing:

```
Building wheels for collected packages: fastavro
  Building wheel for fastavro (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/***/venv/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/tmpm2_a8yuj
       cwd: /private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-install-gshr49e3/fastavro_796163df432e41d1aa0e51de0db4dfca
  Complete output (150 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-universal2-cpython-39
  creating build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/_schema_common.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/_schema_py.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/_logical_writers_py.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/json_read.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/write.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/_write_common.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/_write_py.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/__init__.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/_read_py.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/types.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/json_write.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/_read_common.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/_validate_common.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/_logical_readers_py.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/utils.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/logical_writers.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/_validation_py.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/__main__.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/logical_readers.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/const.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/schema.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/read.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  copying fastavro/validation.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  creating build/lib.macosx-10.9-universal2-cpython-39/fastavro/io
  copying fastavro/io/binary_decoder.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro/io
  copying fastavro/io/__init__.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro/io
  copying fastavro/io/binary_encoder.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro/io
  copying fastavro/io/parser.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro/io
  copying fastavro/io/symbols.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro/io
  copying fastavro/io/json_encoder.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro/io
  copying fastavro/io/json_decoder.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro/io
  creating build/lib.macosx-10.9-universal2-cpython-39/fastavro/repository
  copying fastavro/repository/__init__.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro/repository
  copying fastavro/repository/flat_dict.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro/repository
  copying fastavro/repository/base.py -> build/lib.macosx-10.9-universal2-cpython-39/fastavro/repository
  copying fastavro/py.typed -> build/lib.macosx-10.9-universal2-cpython-39/fastavro
  running build_ext
  
  Error compiling Cython file:
  ------------------------------------------------------------
  ...
              writer_schema, named_schemas, offset, size, return_record_name, return_record_name_override
          )
  
  
  class Block:
      def __init__(
      ^
  ------------------------------------------------------------
  
  fastavro/_read.pyx:976:4: Compiler crash in AnalyseDeclarationsTransform
  
  File 'ModuleNode.py', line 203, in analyse_declarations: ModuleNode(_read.pyx:1:0,
      doc = 'Python code for reading AVRO files',
      full_module_name = 'fastavro._read')
  File 'Nodes.py', line 393, in analyse_declarations: StatListNode(_read.pyx:10:0)
  File 'Nodes.py', line 393, in analyse_declarations: StatListNode(_read.pyx:975:0)
  File 'Nodes.py', line 5121, in analyse_declarations: PyClassDefNode(_read.pyx:975:0,
      name = 'Block')
  File 'Nodes.py', line 393, in analyse_declarations: StatListNode(_read.pyx:976:4)
  File 'Nodes.py', line 2710, in analyse_declarations: CFuncDefNode(_read.pyx:976:4,
      args = [...]/11,
      modifiers = [...]/0,
      outer_attrs = [...]/2,
      overridable = True,
      visibility = 'private')
  File 'Nodes.py', line 2721, in declare_cpdef_wrapper: CFuncDefNode(_read.pyx:976:4,
      args = [...]/11,
      modifiers = [...]/0,
      outer_attrs = [...]/2,
      overridable = True,
      visibility = 'private')
  File 'Nodes.py', line 2787, in call_self_node: CFuncDefNode(_read.pyx:976:4,
      args = [...]/11,
      modifiers = [...]/0,
      outer_attrs = [...]/2,
      overridable = True,
      visibility = 'private')
  
  Compiler crash traceback from this point on:
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/Cython/Compiler/Nodes.py", line 2787, in call_self_node
      type_entry = self.type.args[0].type.entry
  AttributeError: 'PyObjectType' object has no attribute 'entry'
  Compiling fastavro/_read.pyx because it changed.
  [1/1] Cythonizing fastavro/_read.pyx
  Traceback (most recent call last):
    File "/Users/***/venv/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 349, in <module>
      main()
    File "/Users/***/venv/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 331, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/Users/***/venv/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 248, in build_wheel
      return _build_backend().build_wheel(wheel_directory, config_settings,
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 416, in build_wheel
      return self._build_with_temp_dir(['bdist_wheel'], '.whl',
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 401, in _build_with_temp_dir
      self.run_setup()
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 338, in run_setup
      exec(code, locals())
    File "<string>", line 37, in <module>
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/__init__.py", line 107, in setup
      return distutils.core.setup(**attrs)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 185, in setup
      return run_commands(dist)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
      dist.run_commands()
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
      self.run_command(cmd)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/dist.py", line 1234, in run_command
      super().run_command(command)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
      cmd_obj.run()
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/wheel/bdist_wheel.py", line 346, in run
      self.run_command("build")
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
      self.distribution.run_command(command)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/dist.py", line 1234, in run_command
      super().run_command(command)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
      cmd_obj.run()
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/command/build.py", line 131, in run
      self.run_command(cmd_name)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
      self.distribution.run_command(command)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/dist.py", line 1234, in run_command
      super().run_command(command)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
      cmd_obj.run()
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/command/build_ext.py", line 84, in run
      _build_ext.run(self)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/command/build_ext.py", line 345, in run
      self.build_extensions()
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/command/build_ext.py", line 467, in build_extensions
      self._build_extensions_serial()
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/_distutils/command/build_ext.py", line 493, in _build_extensions_serial
      self.build_extension(ext)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/setuptools/command/build_ext.py", line 246, in build_extension
      _build_ext.build_extension(self, ext)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/Cython/Distutils/build_ext.py", line 122, in build_extension
      new_ext = cythonize(
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/Cython/Build/Dependencies.py", line 1134, in cythonize
      cythonize_one(*args)
    File "/private/var/folders/lb/d7gdj0xs0hx502f2y__8x0gc0000gn/T/pip-build-env-4xnztzki/overlay/lib/python3.9/site-packages/Cython/Build/Dependencies.py", line 1301, in cythonize_one
      raise CompileError(None, pyx_file)
  Cython.Compiler.Errors.CompileError: fastavro/_read.pyx
  ----------------------------------------
  ERROR: Failed building wheel for fastavro
Failed to build fastavro
ERROR: Could not build wheels for fastavro which use PEP 517 and cannot be installed directly
```
I have docker file with base image `python:3.9-buster` and this version is working fine on the docker.

**Comment by mdwelsh** [comment]
Looks like this is fixed by #273. But, the Cohere team needs to push a new version to pip with this fix.

**Comment by lfayoux** [comment]
Sorry for the delay, should be updated in 4.19.2
